docs/fundamentals/executor/index.md

```diff
@@ -1,48 +1,38 @@
 (executor-cookbook)=
 # Executor
 
-{class}`~jina.Executor` represents a processing component in a Jina Flow. It performs a single task on a `Document` or 
-`DocumentArray`. 
+{class}`~jina.Executor` represents a processing component in a Jina Flow. It performs a single task on a `DocumentArray`. 
 
 You can create an Executor by extending the `Executor` class and adding logic to endpoint methods.
 
 
+## Why to use Executor?
+
+Once you have learned `DocumentArray`, you are capable of using all its power and expressiveness to build a neural search application.
+Now you need to organize your code in independent modules to be served and scaled independently as microservices.
+
+- Executor lets the developer organize their `DocumentArray` based functions into logical entities that can share configuration state following OOP.
+
+- Executor converts your local functions into functions that can be distributed inside a Flow.
+
+- Using Executor inside a Flow multiple `DocumentArray` can be processed at the same time in a concurrent manner, and deployed easily in the cloud as part of a neural search application.
```

Comment:
```suggestion
- Executors inside a Flow can process multiple `DocumentArray`s simultaneously and concurrently, and can be deployed easily to the cloud as part of your neural search application.
```

Reply:
Let's skip simultaneously , just stick to concurrently here @JohannesMessner , I like the other part of the rewrite however.